core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java

```diff
@@ -148,6 +148,9 @@ public void validate() {
          throw new CacheConfigurationException("It is required to specify a 'failurePolicyClass' when using a " +
                                                 "custom backup failure policy!");
       }
+      if (useTwoPhaseCommit && strategy == BackupConfiguration.BackupStrategy.ASYNC) {
+         throw new CacheConfigurationException("'two-phase-commit' can be used only with SYNC backup strategy!");
```

Comment:
I'm not sure the check is necessary and/or backwards compatible. The `useTwoPhaseCommit()` javadoc doesn't say anything about backup strategies, but the XSD documentation currently says `two-phase-commit` is ignored with ASYNC. At the very least, the documentation should be updated.

Also, the error message should probably be in the `Log` interface.